cmd/openstack-populator/openstack-populator.go

```diff
@@ -0,0 +1,209 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/imagedata"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+const (
+	prefix     = "forklift.konveyor.io"
+	mountPath  = "/mnt/"
+	devicePath = "/dev/block"
+)
+
+const (
+	groupName  = "forklift.konveyor.io"
+	apiVersion = "v1beta1"
+	kind       = "OpenstackVolumePopulator"
+	resource   = "openstackvolumepopulators"
+)
+
+func main() {
+	var (
+		identityEndpoint string
+		imageID          string
+		crNamespace      string
+		crName           string
+		secretName       string
+
+		volumePath string
+	)
+
+	klog.InitFlags(nil)
+
+	// Main arg
+	flag.StringVar(&identityEndpoint, "endpoint", "", "endpoint URL (https://openstack.example.com:5000/v2.0)")
+	flag.StringVar(&secretName, "secret-name", "", "secret containing OpenStack credentials")
+
+	flag.StringVar(&imageID, "image-id", "", "Openstack image ID")
+	flag.StringVar(&volumePath, "volume-path", "", "Path to populate")
+	flag.StringVar(&crName, "cr-name", "", "Custom Resource instance name")
+	flag.StringVar(&crNamespace, "cr-namespace", "", "Custom Resource instance namespace")
+
+	flag.Parse()
+
+	populate(crName, crNamespace, volumePath, identityEndpoint, secretName, imageID)
+}
+
+type openstackConfig struct {
+	username    string
+	password    string
+	domainName  string
+	projectName string
+	insecure    string
+	region      string
+}
+
+func loadConfig(secretName, endpoint, namespace string) openstackConfig {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+
+	klog.Info("Looking for secret", "secret", secretName, "namespace", namespace)
+	secret, err := clientset.CoreV1().Secrets(namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
```

Comment:
we'll need a service account, right? better to change it as we do for ovirt-populator now